common/spec/dependabot/clients/azure_spec.rb

```diff
@@ -66,6 +66,18 @@
         expect { subject }.to raise_error(Dependabot::Clients::Azure::NotFound)
       end
     end
+
+    context "when response body does not contain commit details" do
+      before do
+        stub_request(:get, branch_url).
+          with(basic_auth: [username, password]).
+          to_return(status: 400, body: fixture("azure", "branch_not_found.json"))
```

Comment:
Is that the actual response we get back? I'd expect the body to be empty in this case? I'd also expect the API to return a 404 though so this might very well be what happens 😄

Reply:
If the branch does not exist, the body actually contains exception details. Yeah agreed its kind of weird that it returns 400 and not 404. Let me update the json response file to reflect the actual response body that we get

Reply:
Might be easier to just go by the status code instead of inspecting the body?

Reply:
Agreed status code would be easier but i thought that checking for commit and then raising exception if not found made more sense since the function is trying to fetch commit details. Thoughts?

Reply:
I think that's fine, the branch name is part of the request we're making 👍

Reply:
Cool then i'll update it to raise an error when status code is 400